runelite-client/src/main/java/net/runelite/client/plugins/timetracking/TimeTrackingPlugin.java

```diff
@@ -108,18 +121,44 @@ protected void startUp() throws Exception
 
 		pluginToolbar.addNavigation(navButton);
 
+		timer = new Timer();
+		timer.scheduleAtFixedRate(new TimerTask()
+		{
+			@Override
+			public void run()
+			{
+				updatePanel();
```

Comment:
We do have a `@Schedule` (https://static.runelite.net/api/runelite-client/net/runelite/client/task/Schedule.html) although I do not think it provides sub-tick accuracy currently. I would either maybe modify that, or at least inject the scheduler and schedule it with that. Each java `Timer` is always backed by a java `Thread` which is not great.

Updating swing panels often is a performance issue and even now we have users report that when the farming panel is open (thus, when it updates) it causes noticeable performance issues. Can we make this not update if there are no timesr running? Or if it isn't open?

Reply:
Even if I decrease the tick interval of the `Scheduler`, I can't currently use it because `Hooks#clientMainLoop` does not run when idling on the login screen. 

Only 1 `Timer` is created and used when the plugin is active, which imo is insignificant in terms of resource consumption. (If and when another plugin requires a timer, it can perhaps be moved into `Scheduler` or another class.)

None of the tabs and their contents are being updated when the tab isn't open. (Previously, all patch panels were being updated when updating the UI.) I have also updated it such that while on the clock tab, no UI updates are being done if there are no active clocks.

Reply:
I would maybe prefer if you left the scheduler alone and scheduled it on the `ScheduledExecutorService` which you can inject.